src/LiveComponent/src/DependencyInjection/Compiler/OptionalDependencyPass.php

```diff
@@ -30,8 +31,14 @@ final class OptionalDependencyPass implements CompilerPassInterface
     public function process(ContainerBuilder $container): void
     {
         if ($container->hasDefinition('doctrine')) {
+            $doctrineArguments = [new IteratorArgument([new Reference('doctrine')])]; // TODO: add support for multiple entity managers
             $container->register('ux.live_component.doctrine_entity_hydration_extension', DoctrineEntityHydrationExtension::class)
-                ->setArguments([new IteratorArgument([new Reference('doctrine')])]) // TODO: add support for multiple entity managers
+                ->setArguments($doctrineArguments)
+                ->addTag(LiveComponentBundle::HYDRATION_EXTENSION_TAG)
+            ;
+
+            $container->register('ux.live_component.array_collection_hydration_extension', ArrayCollectionHydrationExtension::class)
```

Comment:
`ux.live_component.doctrine_array..... `  feels a bit more consistent with the other one

Reply:
Done !